extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilder.scala

```diff
@@ -65,8 +65,12 @@ object PrivilegesBuilder {
       if (projectionList.isEmpty) {
         privilegeObjects += PrivilegeObject(table, plan.output.map(_.name))
       } else {
+        val planOutputMapping = plan.output.map(e => e.exprId -> e.name).toMap
```

Comment:
1. Construct a standard dictionary
2. Dereferenced `NamedExpression` by `exprId`
3. Get the column name of the original table by querying the dictionary with `exprId`